### PR TITLE
DOC: add small guide on how to write examples that pass doctests

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Some initial attention points for getting doctests passing (this would eventually fit in the docstring guide, putting it in contributing.rst for now, will move once the other is merged).

Anybody else thinks of typical gotcha's?

cc @datapythonista

### CODE REVIEW

File: doc/source/contributing.rst [hunk 1]
```diff
@@ -430,6 +430,70 @@ the documentation are also built by Travis-CI. These docs are then hosted `here
 <http://pandas-docs.github.io/pandas-docs-travis>`__, see also
 the :ref:`Continuous Integration <contributing.ci>` section.
 
+
```
Comment (jreback) [inline doc/source/contributing.rst]: can you add sub-section refs

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by datapythonista** [comment]
One that comes to my mind is reading from files, like `pandas.read_csv`, that I assume it fails, as the file won't be found.

**Comment by jorisvandenbossche** [comment]
> One that comes to my mind is reading from files, like pandas.read_csv, that I assume it fails, as the file won't be found.

The question is what to do in general for those IO functions. 
Do we want to show actual output? But then there needs to be an actual file to read in, as otherwise it also does not say much. 

In principle you could always create a small file to then read back in.

**Comment by jorisvandenbossche** [comment]
In general, for read functions, we could do like:

```
For the example, we first create a small example csv file

>>> df = pd.DataFrame(...)
>>> df.to_csv('file.csv')

We can now read that in with read_csv:

>>> pd.read_csv('file.csv')

read with option:

>>> ...
```

for write functions that is a bit more difficult, as you don't directly see what you get. Maybe just showing the basic usage (method call) and then referring to user guide is enough for those?

**Review by jreback — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
This was included in the merge of https://github.com/pandas-dev/pandas/pull/19704, so closing.

**Pull request closed without merging**